### PR TITLE
Listcountis 2281085

### PR DESCRIPTION
**Pull request by jesseday**
see: https://drupal.org/node/2281085
Port "List count comparison (list_count_is)" condition to 8.x

Creates ListCountIs plugin and ListCountIsTest class.

Still need to be able to set default_value, options_list and restriction for the operator parameter.

### CODE REVIEW

File: src/Plugin/Condition/ListCountIs.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    switch ($op) {
+      case '==':
+        return count($list) == $value;
+      case '<';
+        return count($list) < $value;
+      case '>';
+        return count($list) > $value;
+    }
+  }
+}
```
Comment (fubhy) [inline src/Plugin/Condition/ListCountIs.php:72]: Missing carriage return at end of file.

File: src/Tests/Condition/ListCountIsTest.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    $this->assertFalse($condition->evaluate());
+
+    // Test that the list count is not equal to 0.
+    $condition = $this->conditionManager->createInstance('rules_list_count_is')
+      ->setContextValue('list', array('one', 'two', 'three'))
+      ->setContextValue('op', '==')
+      ->setContextValue('value', '0');
+    $this->assertFalse($condition->evaluate());
+  }
+}
```
Comment (fubhy) [inline src/Tests/Condition/ListCountIsTest.php:90]: Missing carriage return at end of file. (You can configure your IDE to automatically add that).

File: src/Tests/Condition/ListCountIsTest.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+/**
+ * @file
+ * Contains \Drupal\rules\Tests\Condition\ListCountIsTest.
+ */
+
+namespace Drupal\rules\Tests\Condition;
+
+use Drupal\rules\Plugin\Condition\ListCountIs;
+use Drupal\system\Tests;
```
Comment (fubhy) [inline src/Tests/Condition/ListCountIsTest.php:11]: That use statement seems to be unused.

File: src/Plugin/Condition/ListCountIs.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      ->setDescription(t('The count to compare the data count with.'));
+
+    return $contexts;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summary() {
+    return $this->t('List Count Comparison');
```
Comment (fubhy) [inline src/Plugin/Condition/ListCountIs.php:52]: Shouldn't that be "List count comparison"?

File: src/Plugin/Condition/ListCountIs.php [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+ *
+ */
+class ListCountIs extends RulesConditionBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function contextDefinitions(TypedDataManager $typed_data_manager) {
+    $contexts['list'] = ContextDefinition::create($typed_data_manager, 'list')
+      ->setLabel('List')
```
Comment (fubhy) [inline src/Plugin/Condition/ListCountIs.php:30]: Misses the t(). (Should be t('List'))

File: src/Plugin/Condition/ListCountIs.php [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+use Drupal\Core\TypedData\TypedDataManager;
+use Drupal\rules\Context\ContextDefinition;
+use Drupal\rules\Engine\RulesConditionBase;
+
+/**
+ * Provides a 'List Count Comparison' condition.
+ *
+ * @Condition(
+ *   id = "rules_list_count_is",
+ *   label = @Translation("List Count Comparison")
```
Comment (fubhy) [inline src/Plugin/Condition/ListCountIs.php:19]: Should probably be "List count comparison".

File: src/Plugin/Condition/ListCountIs.php [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+ */
+
+namespace Drupal\rules\Plugin\Condition;
+
+use Drupal\Core\TypedData\TypedDataManager;
+use Drupal\rules\Context\ContextDefinition;
+use Drupal\rules\Engine\RulesConditionBase;
+
+/**
+ * Provides a 'List Count Comparison' condition.
```
Comment (fubhy) [inline src/Plugin/Condition/ListCountIs.php:15]: Should probably be "List count comparison".

**Comment by fubhy** [comment]
Hey! Are you still working on this? If not, I can take over if you want :)

**Comment by jesseday** [comment]
Hey, I'd love to keep working on it, but I'm swamped with catch up work after drupal con.  If you have time to take it over, that would probably be faster than waiting for me at the moment.

**Comment by paranojik** [comment]
Hey @fubhy, @jesseday ... can I take a look?

**Comment by klausi** [comment]
@paranojik sure, just go ahead!

**Comment by paranojik** [comment]
New pull request here: https://github.com/fago/rules/pull/73.

**Pull request closed without merging**